panics upon requesting a doc with an invalid cid

1. open `TestQueryOneToManyWithCidAndDocKey` test in `tests/integration/query/one_to_many/with_cid_dockey_test.go`
```graphql
query {
    book (
        cid: "bafybeictbaivadgsndtnzopt72z446mmg35xvlof45dujfudkjzhwtofq",
        dockey: "bae-fd541c25-229e-5280-b44b-e5c2af3e374d"
    ) {
        name
        author {
            name
        }
    }
}
```
2. alter cid to any value and run

Stack trace:
```
panic: interface conversion: interface {} is nil, not []map[string]interface {} [recovered]
	panic: interface conversion: interface {} is nil, not []map[string]interface {}

goroutine 51 [running]:
testing.tRunner.func1.2({0x105948ca0, 0x14001053f50})
	/opt/homebrew/Cellar/go/1.20.2/libexec/src/testing/testing.go:1526 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.20.2/libexec/src/testing/testing.go:1529 +0x364
panic({0x105948ca0, 0x14001053f50})
	/opt/homebrew/Cellar/go/1.20.2/libexec/src/runtime/panic.go:884 +0x1f4
github.com/sourcenetwork/defradb/tests/integration.assertRequestResults({0x105b6fc10, 0x140001a6008}, 0x140001a6008?, {0x105502316, 0x3d}, 0x140010527b0, {0x140001a3ec8, 0x1, 0x1}, {0x0?, ...}, ...)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:1097 +0x548
github.com/sourcenetwork/defradb/tests/integration.executeRequest({0x105b6fc10, 0x140001a6008}, 0x3d?, {0x140001a3f48?, 0x4?, 0x4?}, {{0x105502316, 0x3d}, {0x140020a9780, 0x4, ...}}, ...)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:935 +0x12c
github.com/sourcenetwork/defradb/tests/integration.executeTestCase({0x105b6fc10, 0x140001a6008}, 0x1400054f380, {0x14000bebe58, 0x2, 0x2}, {{0x105502316, 0x3d}, {0x140020a9780, 0x4, ...}}, ...)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:380 +0xac0
github.com/sourcenetwork/defradb/tests/integration.ExecuteTestCase(0x105a4fd60?, {0x14000bebe58, 0x2, 0x2}, {{0x105502316, 0x3d}, {0x140020a9780, 0x4, 0x4}})
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:288 +0x2f4
github.com/sourcenetwork/defradb/tests/integration.ExecuteRequestTestCase(0xf?, {0x105513ed4?, 0x0?}, {0x14000088e58, 0x2, 0x2}, {{0x105502316, 0x3d}, {0x105515ecc, 0xdc}, ...})
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils.go:134 +0x684
github.com/sourcenetwork/defradb/tests/integration/query/one_to_many.executeTestCase(...)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/query/one_to_many/utils.go:37
github.com/sourcenetwork/defradb/tests/integration/query/one_to_many.TestQueryOneToManyWithCidAndDocKey(0x0?)
	/Users/islam/dev/sourcenetwork/defradb/tests/integration/query/one_to_many/with_cid_dockey_test.go:108 +0x314
testing.tRunner(0x1400054f380, 0x105b57f50)
	/opt/homebrew/Cellar/go/1.20.2/libexec/src/testing/testing.go:1576 +0x104
created by testing.(*T).Run
	/opt/homebrew/Cellar/go/1.20.2/libexec/src/testing/testing.go:1629 +0x370
FAIL	github.com/sourcenetwork/defradb/tests/integration/query/one_to_many	0.541s
FAIL
```
Stack trace before reading from badger storage before the panic:
```
github.com/dgraph-io/badger/v3.(*Txn).Get (/Users/islam/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.5/txn.go:478)
github.com/sourcenetwork/defradb/datastore/badger/v3.(*txn).has (/Users/islam/dev/sourcenetwork/defradb/datastore/badger/v3/datastore.go:642)
github.com/sourcenetwork/defradb/datastore/badger/v3.(*txn).Has (/Users/islam/dev/sourcenetwork/defradb/datastore/badger/v3/datastore.go:638)
github.com/sourcenetwork/defradb/datastore.(*wrappedStore).Has (/Users/islam/dev/sourcenetwork/defradb/datastore/wrappedStore.go:42)
github.com/sourcenetwork/defradb/db.(*db).initialize (/Users/islam/dev/sourcenetwork/defradb/db/db.go:178)
github.com/sourcenetwork/defradb/db.newDB (/Users/islam/dev/sourcenetwork/defradb/db/db.go:125)
github.com/sourcenetwork/defradb/db.NewDB (/Users/islam/dev/sourcenetwork/defradb/db/db.go:93)
github.com/sourcenetwork/defradb/tests/integration.NewBadgerMemoryDB (/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:178)
github.com/sourcenetwork/defradb/tests/integration.GetDatabase (/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:243)
github.com/sourcenetwork/defradb/tests/integration.getStartingNodes (/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:603)
github.com/sourcenetwork/defradb/tests/integration.executeTestCase (/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:309)
github.com/sourcenetwork/defradb/tests/integration.ExecuteTestCase (/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils2.go:288)
github.com/sourcenetwork/defradb/tests/integration.ExecuteRequestTestCase (/Users/islam/dev/sourcenetwork/defradb/tests/integration/utils.go:134)
github.com/sourcenetwork/defradb/tests/integration/query/one_to_many.executeTestCase (/Users/islam/dev/sourcenetwork/defradb/tests/integration/query/one_to_many/utils.go:37)
github.com/sourcenetwork/defradb/tests/integration/query/one_to_many.TestQueryOneToManyWithCidAndDocKey (/Users/islam/dev/sourcenetwork/defradb/tests/integration/query/one_to_many/with_cid_dockey_test.go:108)
testing.tRunner (/opt/homebrew/Cellar/go/1.20.2/libexec/src/testing/testing.go:1576)
testing.(*T).Run.func1 (/opt/homebrew/Cellar/go/1.20.2/libexec/src/testing/testing.go:1629)
runtime.goexit (/opt/homebrew/Cellar/go/1.20.2/libexec/src/runtime/asm_arm64.s:1172)
```

--- a/tests/integration/utils2.go
+++ b/tests/integration/utils2.go
@@ -1145,7 +1145,7 @@ func AssertError(t *testing.T, description string, err error, expectedError stri
 	}
 
 	if expectedError == "" {
-		assert.NoError(t, err, description)
+		require.NoError(t, err, description)
 		return false
 	} else {
 		if !strings.Contains(err.Error(), expectedError) {
@@ -1165,7 +1165,7 @@ func AssertErrors(
 	expectedError string,
 ) bool {
 	if expectedError == "" {
-		assert.Empty(t, errs, description)
+		require.Empty(t, errs, description)
 	} else {
 		for _, e := range errs {
 			// This is always a string at the moment, add support for other types as and when needed

